gio/src/file.rs

```diff
@@ -1111,6 +1113,137 @@ pub trait FileExtManual: IsA<File> + Sized {
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+// checker-ignore-item
+pub enum FileAttributeTypeValue<'a> {
+    Pointer(FileAttributeType, glib::ffi::gpointer),
+    String(&'a str),
+    ByteString(&'a str),
+    Boolean(bool),
+    Uint32(u32),
+    Int32(i32),
+    Uint64(u64),
+    Int64(i64),
+    Object(&'a glib::Object),
+    Stringv(Vec<&'a str>),
+}
+
+impl FileAttributeTypeValue<'_> {
+    pub fn type_(&self) -> FileAttributeType {
+        match self {
+            Self::Pointer(type_, _) => *type_,
+            Self::String(_) => FileAttributeType::String,
+            Self::ByteString(_) => FileAttributeType::ByteString,
+            Self::Boolean(_) => FileAttributeType::Boolean,
+            Self::Uint32(_) => FileAttributeType::Uint32,
+            Self::Int32(_) => FileAttributeType::Int32,
+            Self::Uint64(_) => FileAttributeType::Uint64,
+            Self::Int64(_) => FileAttributeType::Int64,
+            Self::Object(_) => FileAttributeType::Object,
+            Self::Stringv(_) => FileAttributeType::Stringv,
+        }
+    }
+}
+
+impl<'a> ToGlibPtr<'a, glib::ffi::gpointer> for FileAttributeTypeValue<'a> {
+    type Storage = FileAttributeTypeValueStorage<'a>;
+
+    fn to_glib_none(&'a self) -> Stash<'a, glib::ffi::gpointer, Self> {
+        let s = match self {
+            Self::Pointer(_, pointer) => FileAttributeTypeValueStorage::Pointer(*pointer),
+            Self::String(value) => FileAttributeTypeValueStorage::String(
+                ToGlibPtr::<*mut libc::c_char>::to_glib_none(value).1,
+            ),
+            Self::ByteString(value) => FileAttributeTypeValueStorage::ByteString(
+                ToGlibPtr::<*mut libc::c_char>::to_glib_none(value).1,
+            ),
+            Self::Boolean(value) => FileAttributeTypeValueStorage::Boolean(value.into_glib()),
+            Self::Uint32(value) => FileAttributeTypeValueStorage::Uint32(value),
+            Self::Int32(value) => FileAttributeTypeValueStorage::Int32(value),
+            Self::Uint64(value) => FileAttributeTypeValueStorage::Uint64(value),
+            Self::Int64(value) => FileAttributeTypeValueStorage::Int64(value),
+            Self::Object(value) => {
+                FileAttributeTypeValueStorage::Object(ToGlibPtr::<
+                    <glib::Object as glib::translate::GlibPtrDefault>::GlibType,
+                >::to_glib_none(value))
+            }
+            Self::Stringv(value) => FileAttributeTypeValueStorage::Stringv(
+                ToGlibContainerFromSlice::<*mut *mut libc::c_char>::to_glib_none_from_slice(value)
+                    .1,
+            ),
+        };
+        Stash(s.as_ptr(), s)
```

Comment:
Here you get the pointer into `s` (stored on this functions stack), then return (i.e. move) `s` and afterwards the pointer points to this stack frame still while `s` is somewhere else